### ISSUE TITLE
cleanup_test.TestCleanup.test_drop_table_during_cleanup fails due to seastar::gate_closed_exception logged in logging message

### ISSUE THREAD

**Issue by tchaikov** [scylladb/scylladb#16305]
we drop tables while performing cleanup to them in parallel. but there are chances that the operation hits a table whose `gate` is closed, and hence the error in logging message.

```
failed on teardown with "AssertionError: Unexpected errors found: [('node1', ['ERROR 2023-12-06 06:42:23,055 [shard 0:strm] api - force_keyspace_cleanup: keyspace=ks tables={table{name=cf2, id=91e455d0-9402-11ee-81cc-18cc756a1466}, table{name=cf1, id=88f5f550-9402-11ee-81cc-18cc756a1466}, table{name=cf0, id=80d60860-9402-11ee-81cc-18cc756a1466}} failed: seastar::gate_closed_exception (gate closed)', 'ERROR 2023-12-06 06:42:23,003 [shard 0:strm] task_manager - Failed force_keyspace_cleanup of ks.cf0: seastar::gate_closed_exception (gate closed)'])]"
...
```
see also https://jenkins.scylladb.com/job/scylla-master/job/next/6900/testReport/junit/cleanup_test/TestCleanup/Build___x86___dtest___test_drop_table_during_cleanup/

**Comment by tchaikov** [comment]
spotted again at https://jenkins.scylladb.com/job/scylla-master/job/next/6913/

**Comment by bhalevy** [comment]
Hit in https://jenkins.scylladb.com//job/scylla-master/job/next/6913/testReport/junit/cleanup_test/TestCleanup/Build___x86___dtest___test_drop_table_during_cleanup

**Comment by benipeled** [comment]
Similar on https://jenkins.scylladb.com/view/master/job/scylla-master/job/next/6919/testReport/junit/cleanup_test/TestCleanup/Build___x86___dtest___test_drop_table_during_cleanup_2/

`
failed on teardown with "AssertionError: Unexpected errors found: [('node1', ['ERROR 2023-12-10 12:41:49,093 [shard 0:strm] task_manager - Failed force_keyspace_cleanup of ks.cf0: seastar::gate_closed_exception (gate closed)', 'ERROR 2023-12-10 12:41:49,093 [shard 0:strm] api - force_keyspace_cleanup: keyspace=ks tables={table{name=cf2, id=7229f560-9759-11ee-b1c9-99c34ed95232}, table{name=cf1, id=68fb5790-9759-11ee-b1c9-99c34ed95232}, table{name=cf0, id=60d17f90-9759-11ee-b1c9-99c34ed95232}} failed: seastar::gate_closed_exception (gate closed)', 'ERROR 2023-12-10 12:41:49,093 [shard 1:strm] task_manager - Failed force_keyspace_cleanup of ks.cf0: seastar::gate_closed_exception (gate closed)'])]"
`

**Comment by mykaul** [comment]
@tchaikov - please look at this.

**Comment by tchaikov** [comment]
> @tchaikov - please look at this.

it's does seem like another manifest of this bug.

@bhalevy has a better fix at #16351, so i am assigning this issue to him. will close mine -- it is more like a band aid in comparison with Benny's fix. 

in the mean while, still reviewing his change.
